src/helpers.js

```diff
@@ -424,13 +424,6 @@ export function findHostNodeDevice(hostdev, nodeDevices) {
                 d.capability.vendor.id == vendorId &&
                 d.capability.product.id == productId)
                 return true;
-            else if (vendorId &&
-                productId &&
-                d.capability.vendor &&
-                d.capability.product &&
-                d.capability.vendor.id == vendorId &&
-                d.capability.product.id == productId)
-                return true;
         });
         break;
```

Comment:
@yunmingyang  I'm going to drop this change and reintroduce it back.
According to libvirt xml schema, source bus and device numbers are optional, therefore we should be able to look up host device even without them present: https://gitlab.com/libvirt/libvirt/-/blob/master/docs/schemas/domaincommon.rng#L3292

Reply:
This was the change which was causing the c-machines not being able to look up a correct node device, therefore it could not properly show vendor name, product name and bus and device number. (The problem we talked about on IRC that user could consider a bus and device number missing a bug)

Reply:
But, that will still cause an issue that c-machines filter a wrong device from nodeDevices, and that will also cause deletion error

Reply:
I slightly modified the function, so now it works for both cases: When bus and device number is present, and when it is not present (and there is a risk of 2 node devices sharing same vendor and product ids)